external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -1124,30 +1257,90 @@ fn parse_sequence(context: &mut Context) -> Result<Sequence, Box<Diagnostic>> {
     let mut last_semicolon_loc = None;
     let mut eopt = None;
     while context.tokens.peek() != Tok::RBrace {
-        let item = parse_sequence_item(context)?;
-        if context.tokens.peek() == Tok::RBrace {
-            // If the sequence ends with an expression that is not
-            // followed by a semicolon, split out that expression
-            // from the rest of the SequenceItems.
-            match item.value {
-                SequenceItem_::Seq(e) => {
-                    eopt = Some(Spanned {
-                        loc: item.loc,
-                        value: e.value,
-                    });
+        // this helps when a sequence contains a comma-separated list without the ending token (in
+        // which case the parser would be likely fast-forwarded to EOF)
+        context.stop_set.add(Tok::Semicolon);
+        match parse_sequence_item(context) {
+            Ok(item) => {
+                context.stop_set.remove(Tok::Semicolon);
+                if context.tokens.peek() == Tok::RBrace {
+                    // If the sequence ends with an expression that is not
+                    // followed by a semicolon, split out that expression
+                    // from the rest of the SequenceItems.
+                    match item.value {
+                        SequenceItem_::Seq(e) => {
+                            eopt = Some(Spanned {
+                                loc: item.loc,
+                                value: e.value,
+                            });
+                        }
+                        _ => {
+                            context
+                                .env
+                                .add_diag(*unexpected_token_error(context.tokens, "';'"));
+                        }
+                    }
+                    break;
+                }
+                seq.push(item);
+                last_semicolon_loc = Some(current_token_loc(context.tokens));
+                if let Err(diag) = consume_token(context.tokens, Tok::Semicolon) {
+                    let at_stop_set = skip_sequence(context, *diag);
+                    if at_stop_set {
+                        break;
+                    }
+                }
```

Comment:
This pattern seems to come up a lot, I wonder if its worth macro-ifying or something